### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by fhsevenbel**
This PR adds docker container support

A Dockerfile was created as well as a docker-compose.yml.
After copying the zuliprc and adding the .env variables a simple docker-compose up -d will be able to start the reminder bot
Added a host variable in the src/store.ts file to be able to execute docker container, updated documentation

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+REMINDER_DATA=
```
Comment (ornicar) [inline .env.example]: what is this about?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,52 @@
 dist
 node_modules
 zuliprc
+.env
+
+# Adds generated gitignore for Windows and Linux files
```
Comment (ornicar) [inline .gitignore]: No.

Keep your environment specific gitignores [to your environment](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer)

Only the project-relevant ignores go to the project gitignore.

**Review by ornicar — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (fhsevenbel) [reply .gitignore]: Will be done, thank you i did not know this was possible!

File: .env.example (hunk 1 above)
Comment (fhsevenbel) [reply .env.example]: In the docker-compose.yml file you can see the REMINDER_DATA is used as an environment variable to describe where the volume of the redis database is located on the HOST system

I need this variable to make reminders persistent even if the docker container is destroyed.

Comment (ornicar) [reply .env.example]: ah ok, so maybe rename it to `DOCKER_REDIS_VOLUME` to make its purpose clearer, and so people not using docker know not to worry about it.

Comment (fhsevenbel) [reply .env.example]: has been done

**Pull request merged**